docs/backend/deprecation.md

```diff
@@ -0,0 +1,314 @@
+---
+myst:
+  html_meta:
+    "description": "A guide how to do deprecations, including Python, ZCML and templates in Plone."
+    "property=og:description": "A guide how to do deprecations, including Python, ZCML and templates in Plone."
+    "property=og:title": "Deprecation"
+    "keywords": "deprecation, zcml, template, jbot"
+---
+
+(backend-deprecation-label)=
+# Deprecation
+
+## Introduction
+
+This document describes rationales, configuration and best practices of deprecations in Plone, Zope and Python.
+It is meant as a styleguide on how to apply deprecations in Plone core packages.
+It also has a value as a general overview on how to deprecate in Python.
+
+
+### Why Deprecation
+
+At some point we
+
+- need to get rid of old code,
+- want to unify api style (consistent api),
+- fix typos in namings,
+- move code or templates around (inside package or to another package).
+
+While refactoring code, moving modules, functions, classes and methods is often needed.
+To not break third party code imports from the old place or usage of old functions/ methods must work for while.
+Deprecated methods are usually removed with the next major release of Plone.
+
+Following the [semantic versioning guideline](http://semver.org) is recommended.
+
+### Help Programmers, No annoyance
+
+Deprecation has to support the consumers of the code - the programmers using it.
+From their point of view, Plone core code is an API to them.
+Any change is annoying to them anyway, but they feel better if deprecation warnings are telling them what to do.
+
+Deprecations must always log at level *warning* and have to answers the question:
+
+**"Why is the code gone from the old place? What to do instead?"**
+
+A short message is enough., i.e.:
+
+- "Replaced by new API xyz, found at abc.cde".,
+- "Moved to xyz, because of abc.",
+- "Name had a typo, new name is "xyz".
+
+All logging has to be done once, i.e. on first usage or first import.
+It must not flood the logs.
+
+### Use Cases
+
+Renaming
+
+: We may want to rename classes, methods, functions or global or class variables in order to get a more consistent api or because of a typo, etc.
```

Comment:
```suggestion
: We may want to rename classes, methods, functions or global or class variables in order to get a more consistent API or because of a typo, etc.
```